src/renderer.tsx

```diff
@@ -20,22 +20,29 @@ export interface NotionRendererProps {
   level?: number;
   customBlockComponents?: CustomBlockComponents;
   customDecoratorComponents?: CustomDecoratorComponents;
+
+  strict?: boolean;
 }
 
 export const NotionRenderer: React.FC<NotionRendererProps> = ({
   level = 0,
   currentId,
   mapPageUrl = defaultMapPageUrl,
   mapImageUrl = defaultMapImageUrl,
+  strict = false,
   ...props
 }) => {
   const { blockMap } = props;
   const id = currentId || Object.keys(blockMap)[0];
   const currentBlock = blockMap[id];
 
   if (!currentBlock) {
+    const errorMessage = `error rendering block ${currentId}: block not in blockMap`;
+    if (strict) {
+      throw new Error(errorMessage);
+    }
     if (process.env.NODE_ENV !== "production") {
-      console.warn("error rendering block", currentId);
+      console.warn(errorMessage);
     }
     return null;
```

Comment:
there's at least one other place in the code where you'll want to handle missing block support in the `block` switch / case statement.